akka-cluster/src/main/scala/akka/cluster/ClusterActorRefProvider.scala

```diff
@@ -64,7 +61,31 @@ private[akka] class ClusterActorRefProvider(
    */
   override protected def createDeployer: ClusterDeployer = new ClusterDeployer(settings, dynamicAccess)
 
-  override protected def showDirectUseWarningIfRequired(): Unit = ()
+  override def actorOf(
+      system: ActorSystemImpl,
+      props: Props,
+      supervisor: InternalActorRef,
+      path: ActorPath,
+      systemService: Boolean,
+      deploy: Option[Deploy],
+      lookupDeploy: Boolean,
+      async: Boolean): InternalActorRef = {
+    val ref = super.actorOf(system, props, supervisor, path, systemService, deploy, lookupDeploy, async)
+    ref match {
+      case remoteRef: RemoteActorRef if !remoteSettings.UseUnsafeRemoteFeaturesWithoutCluster =>
+        val remoteAddress = remoteRef.path.address
+        if (Cluster(system).state.members.exists(_.address == remoteAddress))
+          remoteRef
+        else {
+          log.warning(
+            "Remote deploy of [{}] outside this cluster is not allowed, falling back to local.",
+            remoteAddress)
+          local.actorOf(system, props, supervisor, path, systemService, deploy, lookupDeploy, async)
```

Comment:
I'd thought so too, updating. It was originally from a suggestion but I missed it.

Reply:
@raboof Apparently it would not hurt since `start` is not yet called but my last commit removed the need to override actorOf in the first place - safer.